File: platform/backend/v_platform/schemas/system_settings.py
"""시스템 설정 스키마

시스템 전역 설정 API 요청/응답 스키마
"""

from pydantic import BaseModel, Field


class SystemSettingsBase(BaseModel):
    """시스템 설정 기본 스키마"""

    manual_enabled: bool = Field(default=True, description="메뉴얼 링크 표시 여부")
    manual_url: str = Field(default="http://localhost:3000", description="메뉴얼 URL")
    default_start_page: str = Field(
        default="/", description="시스템 기본 시작 페이지 경로"
    )
    # 앱 브랜딩
    app_title: str | None = Field(default=None, description="앱 표시 이름 (TopBar, Login)")
    app_description: str | None = Field(default=None, description="앱 설명 (Login 페이지)")
    app_logo_url: str | None = Field(default=None, description="로고 이미지 URL")


class SystemSettingsUpdate(BaseModel):
    """시스템 설정 업데이트 스키마"""

    manual_enabled: bool | None = None
    manual_url: str | None = None
    default_start_page: str | None = None
    app_title: str | None = None
    app_description: str | None = None
    app_logo_url: str | None = None


class SystemSettingsResponse(SystemSettingsBase):
    """시스템 설정 응답 스키마"""

    id: int

    class Config:
        """Pydantic 설정"""

        from_attributes = True

File: platform/frontend/v-platform-core/src/api/systemSettings.ts
/**
 * 시스템 설정 API 클라이언트
 */

import { apiClient } from "./client";

export interface SystemSettings {
  id: number;
  manual_enabled: boolean;
  manual_url: string;
  default_start_page: string;
  // 앱 브랜딩
  app_title?: string;
  app_description?: string;
  app_logo_url?: string;
}

export interface SystemSettingsUpdate {
  manual_enabled?: boolean;
  manual_url?: string;
  default_start_page?: string;
  app_title?: string;
  app_description?: string;
  app_logo_url?: string;
}

export interface PublicBranding {
  app_title?: string | null;
  app_description?: string | null;
  app_logo_url?: string | null;
}

export const systemSettingsApi = {
  /**
   * 시스템 설정 조회
   */
  getSettings: async (): Promise<SystemSettings> => {
    const response = await apiClient.get<SystemSettings>(
      "/api/system-settings/",
    );
    return response.data;
  },

  /**
   * 시스템 설정 업데이트 (관리자 전용)
   */
  updateSettings: async (
    update: SystemSettingsUpdate,
  ): Promise<SystemSettings> => {
    const response = await apiClient.put<SystemSettings>(
      "/api/system-settings/",
      update,
    );
    return response.data;
  },

  /**
   * 공개 브랜딩 조회 (인증 불필요) - 로그인 페이지용
   */
  getPublicBranding: async (): Promise<PublicBranding> => {
    const response = await apiClient.get<PublicBranding>(
      "/api/system-settings/branding",
    );
    return response.data;
  },
};
